src/calendar/__specs__/calendar_spec.js

```diff
@@ -1,557 +1,337 @@
+        })
+      })
     })
-    const calendarEl = findDOMNode(calendar)
-    const nextLinkEl = calendarEl.querySelector('.calendar-header_button.is-next')
-    const headerTitleEl = calendarEl.querySelector('.calendar-month_header_title')
 
-    assert(headerTitleEl.textContent === 'July 2015')
-    TestUtils.Simulate.click(nextLinkEl)
-    assert(headerTitleEl.textContent === 'August 2015')
-  })
+    context('when nextProps `activeMonth` is `undefined`', () => {
+      it("doesn't set state", () => {
+        props = {activeMonth: date}
+        wrapper = shallow(<Calendar />)
+        wrapper.setProps({disableDaysOfWeek: false})
 
-  it('updates default today when date is changed', () => {
-    // Set current date to 2015-07-05
-    let clock = sinon.useFakeTimers(new Date(2015, 6, 5).getTime())
-    const calendar = render({activeMonth: new Date(2015, 6, 1)})
-    const todayText = () => findDOMNode(calendar).querySelector('.calendar-day.is-today').textContent
-    assert(todayText() === '5')
-    // Change current date to 2015-07-17
-    clock.restore()
-    clock = sinon.useFakeTimers(new Date(2015, 6, 17).getTime())
-    calendar.forceUpdate()
-    assert(todayText() === '17')
-    clock.restore()
+        expect(wrapper.state('activeMonth')).not.toEqual(new Date(2015, 9, 1))
+      })
+    })
   })
 
-  context('in single selection mode', () => {
-    let onSelect, calendar
-    beforeEach(() => {
-      onSelect = sinon.spy()
-      calendar = render({
-        activeMonth: new Date(2015, 5, 1),
-        mode: 'single',
-        onSelect,
+  describe('#_initialMonth', () => {
+    context('when prop `activeMonth` is valid date', () => {
+      it('returns `activeMonth`', () => {
+        wrapper = shallow(<Calendar activeMonth={date} />)
+
+        expect(wrapper.instance()._initialMonth()).toEqual(date)
       })
     })
 
-    it('can select single day', () => {
-      clickOnDays(calendar, 5)
-      assert(onSelect.calledWith(new Date(2015, 5, 6)))
+    context('when prop `activeMonth` is invalid date', () => {
+      it('returns `activeMonth`', () => {
+        props = {
+          activeMonth: '2017-01-01',
+          selected: 'abc',
+          today: date,
+        }
+        wrapper = shallow(<Calendar {...props} />)
+
+        expect(wrapper.instance()._initialMonth()).toEqual(props.today)
+      })
     })
   })
 
-  describe('range selection mode', () => {
-    context('when onSelectionProgress is not passed', () => {
-      let onSelect, calendar
-      beforeEach(() => {
-        onSelect = sinon.spy()
-        calendar = render({
-          activeMonth: new Date(2015, 5, 1),
-          selected: {start: new Date(2015, 5, 1), end: new Date(2015, 5, 3)},
-          mode: 'range',
-          onSelect,
-        })
-      })
+  describe('#_switchMonth', () => {
+    context('when prop `onMonthChange` is defined', () => {
+      it('calls prop #onMonthChange', () => {
+        props = {onMonthChange: jest.fn()}
```

Comment:
I think that `activeMonth` prop is missing here

Tests are passing without a warning if we add:
```js
props = {onMonthChange: jest.fn(), activeMonth: date}
```
<img width="855" alt="screen shot 2018-01-08 at 10 21 27" src="https://user-images.githubusercontent.com/4289338/34664775-b891fbe2-f45d-11e7-8f63-608c9e843b4c.png">